scripts/utils.mjs

```diff
@@ -1,6 +1,6 @@
 import path from 'path';
 import url from 'url';
-import fse from 'fs-extra';
+import fs from 'fs';
```

Comment:
```suggestion
import fs from 'fs/promises';
```

And use the async variant?

Reply:
Is it better to use the async variant? We'll have to `await` for it here: https://github.com/mui/material-ui/pull/43234/files#diff-d3831cd61824ee8713cea7361c0fe4cb921f80deb8eaec5ebccc434be0e96d6eL32

Reply:
Yes, we should always use the non-blocking variants. Just to free up the event loop while node.js is interfacing with the operating system. I think we can treat the sync fs functions as deprecated.

Reply:
Implemented 👌🏼 

This is what I understand: We're giving the opportunity of using `getVersionEnvVariables` in a non-blocking way, even though, in this particular case, we're blocking its usage in `build.mjs` execution as we need the version values before moving forward.

Is that correct?

Reply:
blocking vs. non-blocking is about keeping the main thread busy or not. As an example, the following would act as a blocking timer:

```tsx
function blockingTimer(ms) {
  const end = Date.now() + ms;
  while (Date.now() < end) {}
}
```

This function will act like `setTimout`, except it will block the main thread. Nothing else can run. It is to say, suppose you have an interval running like
```tsx
setInterval(() => console.log('hello'), 1000)
```
and you run `blockingTimer(5000)`. The interval will skip 5 beats.

To apply this to `readFile`, it's important to understand that Node.js is interfacing with the underlying operating system. It opens a file, and reads it into a buffer chunk by chunk. A big part of the work that is done is not happening on the main thread. Node.js can execute other code while it's interfacing with the OS file system. `readFileSync` acts like the `blockingTimer` from above.

The problem of main thread blocking code is when multiple things are being done in parallel. 

```tsx
await Promise.all([
  getVersionEnvVariables(),
  doOtherAsyncWork(),
])
```

If we use `readFileSync` under the hood, then even though we try to run both functions in parallel, they would essentially execute in series as `getVersionEnvVariables()` blocks the main thread most of the time.

Reply:
Thanks for the detailed explanation @Janpot